File: app/src/main/java/com/savvasdalkitsis/gameframe/model/DisplayMode.java
package com.savvasdalkitsis.gameframe.model;

public enum DisplayMode {

    GALLERY("m0", 0),
    CLOCK("m1", 1);

    private String queryParamName;
    private int level;

    DisplayMode(String queryParamName, int level) {
        this.queryParamName = queryParamName;
        this.level = level;
    }

    public String getQueryParamName() {
        return queryParamName;
    }

    public static DisplayMode from(int level) {
        for (DisplayMode brightness : values()) {
            if (brightness.level == level) {
                return brightness;
            }
        }
        return level < 0 ? GALLERY : CLOCK;
    }
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/control/model/DisplayMode.kt
package com.savvasdalkitsis.gameframe.control.model

enum class DisplayMode(val queryParamName: String, private val level: Int) {

    GALLERY("m0", 0),
    CLOCK("m1", 1);


    companion object {

        fun from(level: Int): DisplayMode {
            return values().firstOrNull { it.level == level }
                    ?: if (level < 0) GALLERY else CLOCK
        }
    }
}
